setup.cfg

```diff
@@ -1,21 +0,0 @@
-[flake8]
-ignore = W503, E203, B305
-max-line-length = 88
-
-[mypy]
-disallow_untyped_defs = True
-ignore_missing_imports = True
-
-[tool:isort]
-combine_as_imports = True
-force_grid_wrap = 0
-include_trailing_comma = True
-known_first_party = httpcore,tests
-known_third_party = brotli,certifi,chardet,cryptography,h11,h2,hstspreload,pytest,rfc3986,setuptools,sniffio,trio,trustme,urllib3,uvicorn
-line_length = 88
-multi_line_output = 3
-
-[tool:pytest]
-addopts = --cov=httpcore --cov=tests --cov-report=term-missing -rxXs
-markers =
-  copied_from(source, changes=None): mark test as copied from somewhere else, along with a description of changes made to accodomate e.g. our test setup
```

Comment:
This file is actually quite useful for me as it's read by the editor when linting and configures the tools when I invoke them myself (usually just `pytest` but still). I also find it nicer to have all the tools config in a single, more readable place as some of them can be quite long.

I'd prefer keeping it, if that's ok?

Reply:
Possibly.

I'm not wild about `setup.cfg` myself, and keen on removing top-level config files from the repo where possible, but it's certainly worth a conversation about what we're happy to have across all the encode repos as a whole.

I'm defo happy to drop this from the PR, and keep any changes there scoped to another conversion.

Reply:
Agreed with Yeray on this - these config files are super useful as IDEs that automatically read them without having to change anything in project settings.